book/plugins/android/single-bundle.md

```diff
@@ -0,0 +1,34 @@
+## Single Bundle React Naitve
+
+![single-bundle.png](./single-bundle.png)
+
+### OVERVIEW
+This plugin gives ability to change version of react native and react. It also aggregate all react native plugins into one single bundle and upload it to S3. Plugins becomes modules under the names specified as plugin identifiers from plugin configuration."
+### PLUGINS PREPERATIONS
+
+#### Add Single Bundle Plugin to project configuration
+Add single bundle plugin and select versions of react-native, react and react-native-zapp-bridge you want to use. Save plugin configuration and rebuild. Please note rebuild is necesery every time you change versions or adding/removing other react native plugins.
+
+![Single Bundle Configuration]( https://assets-production.applicaster.com/applicaster-employees/zapp_team/anna_bauza/react_native/single-bundle-config.png  "Single Bundle Configuration")
+
+#### Make sure plugins support selected versions
+Your React Native plugins need to be build and published to NPM for selected versions of react and react-native.
+
+
+#### Migrate your plugins
+In order to make Single Bundle aggregator work please make sure all your react native plugins has been migrated and published to npm as following [Plugins Migration](/react-native/plugins-migration.md) instructions advice.
+
+### RUNNING LOCAL SERVER FOR SINGLE BUNDLE
+
+In Zapp-Android folder execute command :
+`bundle exec rake single_bundle_aggregator:build`.
+Please ignore that rake will abort due to lack of AWS credentials - this is expected for local environment.
+
+Rake process will create a folder `single-bundle-aggregator` - this is your local server for single bundle.
+
+If you wish to debug package inside your single-bundle please place the package folder inside `single-bundle-aggregator/packages`. Please edit `single-bundle-aggregator/index.js` to give information about local packages.
+
+Local single bundle is ready to run.
+
+### PLUGIN MISSING EXPECTED VERSIONS?
+Please contact the Zapp Team.
```

Comment:
@f-roland please note Team Zapp is in charge of versions which are provided.

Reply:
I'm not sure what this means, could you elaborate ?

Reply:
it is related to your question `even with the breaking changes in metro-bundler / babel versions on react 55 / 56 ?` so if you don't want to support these versions we don't.

Reply:
right, I see. 
My point is that today, most of our plugins (if not all) are written for 0.50.4. I know there are big tooling changes which happened somewhere between React 55 or 56, and 58 is adding breaking changes in the scrollview.

this is why I was wondering about the range of version we support - we should test these scenarios, see which want we can / want to cover, then mention them here. Simply saying "pick any RN version you want" might not turn out to be true

I would of course advocate for later versions, but we have to be practical as we won't rewrite all plugins. We sent the plugin migration guide a few months ago, and I'm not sure a lot of plugins were updated already.

@lomudi @liviur do you guys have any recommendation on how to handle this ?

Reply:
ahh right maybe "pick RN version from available list" would work?

Reply:
@f-roland  I've put currently supported versions in description. Please note that using 0.50.4 is crashing on multiple usage of react components due to caching defect on 0.50.4 on Android. All Ran Sport plugins has been written in 0.57.8 which is taken from our proof of concept (aggregator example). Bumping this version fixed this crash on Android.

Reply:
Yes that's the idea 👍